packages/compass-aggregations/src/modules/pipeline-builder/stage-editor.ts

```diff
@@ -453,28 +542,57 @@ export const addStage = (
   after?: number
 ): PipelineBuilderThunkAction<void, StageAddAction> => {
   return (dispatch, getState, { pipelineBuilder }) => {
-    const stage = pipelineBuilder.addStage(after);
+    const {
+      pipelineBuilder: {
+        stageEditor: { stages },
+      },
+    } = getState();
+    const addAfter = after ?? stages.length;
+    const addAfterIdxInPipeline = storeIndexToPipelineIndex(stages, addAfter, {
+      // The item at afterIndex could be a wizard hence we need to consider
+      // that while calculating the corresponding afterIdx in pipeline
+      includeIndex: true,
+    });
+
+    const stage = pipelineBuilder.addStage(addAfterIdxInPipeline);
     track('Aggregation Edited', {
-      num_stages: getState().pipelineBuilder.stageEditor.stages.length,
+      num_stages: pipelineFromStore(stages).length,
       stage_action: 'stage_added',
       stage_index: stage.id + 1,
```

Comment:
I don't this is correct as well. The `stage.id` is an incremental number which does not necessarily corresponds to the stage index. We probably should just use, `addAfterIdxInPipeline  + 1` here

Reply:
Yes, this seems to be wrong. We should fix this

Reply:
Disregarding this like discussed